source/Ship.cpp

```diff
@@ -630,7 +618,15 @@ void Ship::FinishLoading(bool isNewInstance)
 				count -= eit->second;
 
 			if(count)
-				armament.Add(it.first, count);
+				count -= armament.Add(it.first, count);
+			if(count)
```

Comment:
This second if should be in the first's body, yes?

Reply:
Yes, that is nicer, no need to check count again if it was already at zero.